language-plutus-core/test/Evaluation/DynamicBuiltins/Definition.hs

```diff
@@ -120,18 +110,21 @@ test_dynamicId =
             one = mkConstant @Integer @DefaultUni () 1
             integer = mkTyBuiltin @Integer ()
             -- id {integer -> integer} ((\(i : integer) (j : integer) -> i) 1) 0
-            term =
-                mkIterApp () (TyInst () dynamicId (TyFun () integer integer))
-                    [ Apply () constIntegerInteger one
-                    , zer
-                    ] where
-                          constIntegerInteger = runQuote $ do
-                              i <- freshName "i"
-                              j <- freshName "j"
-                              return
-                                  . LamAbs () i integer
-                                  . LamAbs () j integer
-                                  $ Var () i
+            term = Apply () (ApplyBuiltin ()
+                                 (DynBuiltinName dynamicIdName)
+                                 [TyFun () integer integer]
+                                 [Apply () constIntegerInteger one]
+                            )
+                            zer
```

Comment:
What happens if we write

```haskell
ApplyBuiltin ()
    (DynBuiltinName dynamicIdName)
    [TyFun () integer integer]
    [Apply () constIntegerInteger one, zer]
```

? Type error, I guess? "Too many arguments". Is this what we want? Probably yes, but pinging @michaelpj just in case.

Reply:
Too many arguments sounds right, and I expect to get that from the typechecker (which I think we will in this PR). Maybe worth some tests for the over/under application error cases.

Reply:
I did make a mini-issue to add some tests at https://jira.iohk.io/browse/SCP-771 a while ago.  I was going to do that immediately after this is merged (if it ever is merged).